### DISCUSSION TITLE
Specify decimals of numeric variables when using write_sav()

### DISCUSSION THREAD

**Discussion by diego-EA** [Roche/pyreadstat#114] [Q&A] [answered]
Hello everybody, 

I'm trying to write a SPSS file with 700 variables and I have two variables ('g_nps_1' and 'g_nps_2') that require high precision (13 decimals after the comma). When I use the lines below, I only get 2 decimals (being _df_ my data frame and _md_ my metadata object):

`md.original_variable_types['g_nps_1'] = 'F20.13'`
`md.original_variable_types['g_nps_2'] = 'F20.13'`

`pyreadstat.write_sav(df, os.path.join(data_path, output_file), column_labels=md.column_labels,        variable_display_width=md.variable_display_width, variable_measure=md.variable_measure,                      variable_value_labels=md.variable_value_labels)`

I have tried this: 
`formats = {'g_nps_1': 'F20.13', 'g_nps_2': 'F20.13'}`
`pyreadstat.write_sav(df_aux_2, os.path.join(data_path, output_file), variable_format=formats)`

But I get this error: 
_TypeError: write_sav() got an unexpected keyword argument 'variable_format'_

What I am doing wrong? How can I define the format  'F20.13' for these 2 variables? Thanks a lot in advance.

**Comment by ofajardo** [comment, marked as answer]
hmm, this looks correct: 

```
formats = {'g_nps_1': 'F20.13', 'g_nps_2': 'F20.13'}
pyreadstat.write_sav(df_aux_2, os.path.join(data_path, output_file), variable_format=formats)
```

and the error: 

```
TypeError: write_sav() got an unexpected keyword argument 'variable_format'
```

suggests that you are using an old version of pyreadstat (the argument is new in version 1.0.7) ... could you please check what version of pyreadstat you are using (print pyreadstat.\_\_version\_\_) and if old please update it

**Reply by diego-EA** [reply]
That was it, I was using Version 1.0.5. With Version 1.0.9 it works perfectly. Many thanks for the advice!

**Reply by ofajardo** [reply]
Great!
